lib/Doctrine/MongoDB/Query/Expr.php

```diff
@@ -162,6 +162,75 @@ public function all(array $values)
     }
 
     /**
+     * Apply a bitwise operation on the current field
+     *
+     * @see http://docs.mongodb.org/manual/reference/operator/update/bit/
+     * @param string $operator
+     * @param int $value
+     * @return self
+     */
+    protected function bit($operator, $value)
+    {
+        $this->requiresCurrentField();
+        $this->newObj['$bit'][$this->currentField][$operator] = $value;
+        return $this;
+    }
+
+    /**
+     * Apply a bitwise and operation on the current field.
+     *
+     * @see Builder::bitAnd()
+     * @see http://docs.mongodb.org/manual/reference/operator/update/bit/
+     * @param int $value
+     * @return self
+     */
+    public function bitAnd($value)
+    {
+        return $this->bit('and', $value);
+    }
+
+    /**
+     * Apply a bitwise or operation on the current field.
+     *
+     * @see Builder::bitOr()
+     * @see http://docs.mongodb.org/manual/reference/operator/update/bit/
+     * @param int $value
+     * @return self
+     */
+    public function bitOr($value)
+    {
+        return $this->bit('or', $value);
+    }
+
+    /**
+     * Apply a bitwise xor operation on the current field.
+     *
+     * @see Builder::bitXor()
+     * @see http://docs.mongodb.org/manual/reference/operator/update/bit/
+     * @param int $value
+     * @return self
+     */
+    public function bitXor($value)
+    {
+        return $this->bit('xor', $value);
+    }
+
+    /**
+     * Sets the value of the current field to the current date, either as a date or a timestamp.
+     *
+     * @see Builder::currentDate()
+     * @see http://docs.mongodb.org/manual/reference/operator/update/currentDate/
+     * @param bool $useTimestamp
+     * @return self
+     */
+    public function currentDate($useTimestamp = false)
```

Comment:
The `@Timestamp` mapping is basically stuck in ODM now (and I suppose it has some relevance since MongoTimestamp is a legitimate class the driver could return).

There is a similar need if we were to ever introduce constants for `Expr::type()` (see: [here](https://github.com/doctrine/mongodb/blob/c3c029cab1917afb8fa7cc41ef3dc328e8a852fb/lib/Doctrine/MongoDB/Query/Expr.php#L1126), where we currently use hard-coded strings), so any decision to introduce constants should be mindful to use a prefix so not to cause a conflict down the line (e.g. using `CURRENTDATE_DATE` instead of just `DATE`). I don't think we need to introduce constants for this, though. Although `"date"` is a "magic string", it's something 99% of users would be using, so I think we can get by with having it as a default value.

If you're sold on having constants for this, though, I won't object. Just prefix them :)